Endpoint for user leaving the site

Simonlie0302 opened the pull request:
I'm having trouble testing this feature. Currently, the frontend doesn't send a fetch call when the user leaves the page or navigates back to the previous page. It should be fairly straightforward for the backend to handle this as long as the frontend sends the necessary request. Any guidance on how to achieve this?

riccardobellini commented:
The frontend should perform the call when handling the `onbeforeunload` event listener, to be instantiated on the window object. It's better to add this event listener and keep a `ref` of it, in order to dispose it correctly. If we want to track whether a user left the home page, the event listener can be instantiated only inside the Home component.

Note that the event will fire on `beforeunload`, which means also when navigating away to a different URL, not only in response to browser/tab closing. But we can tell (when analysing the data) whether the event was the user actually closing or if it was just a navigation to the review page based on whether the user performed other actions in the db or not.

Simonlie0302 commented:
@Icon1cc I added the setupBeforeUnloadListener() to the home page and now it tracks if a user left the home page, but only the home page.

Simonlie0302 commented:
> @Icon1cc I added the setupBeforeUnloadListener() to the home page and now it tracks if a user left the home page, but only the home page.

I added it for the review section as well and it works fine.

Simonlie0302 commented on frontend/src/pages/ReviewWizard.tsx:
```diff
@@ -58,7 +69,7 @@ const ReviewWizard: React.FC = () => {
 
     if (loading) return <Loader />;
-    if (questions.length === 0) return <Loader />;
+    if (questions.length === 0) return <div>No questions available.</div>;
```
Oh wait, ill have to modify this probably
